### PR TITLE
[Content] New documentation site structure

### PR DESCRIPTION
**Pull request by ronijaakkola**
## Description
This PR changes the new documentation site structure. The new structure is as follows; Getting Started, Foundation, Components, Patterns, About.

## How Has This Been Tested?
Tested by running the documentation site locally.

### CODE REVIEW

File: new-site/src/docs/components/tooltip/code.mdx [hunk 1]
```diff
@@ -73,3 +70,5 @@ Note! You can find the full list of properties in the <Link size="M" href="/stor
 | `buttonLabel`  | Aria-label text for the tooltip trigger button.              | `string`                                                                                                                                                                                       | "Tooltip"     |
 | `tooltipLabel` | Aria-label text for the tooltip.                             | `string`                                                                                                                                                                                       | "Tooltip"     |
 [Table 1:Tooltip properties]|
+
+export default ({ children }) => <>{children}</>;
```
Comment (minevala) [inline new-site/src/docs/components/tooltip/code.mdx:74]: This is export is not needed

File: new-site/src/docs/components/index.mdx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+slug: "/components"
+title: "Components overview"
+---
+
+# Components overview
+
+Some component content
```
Comment (minevala) [inline new-site/src/docs/components/index.mdx:8]: I think we could add the component table here as we have it on old site

**Review by minevala — COMMENTED** [review]
Looks good 👍 I found is one unnecessary export and there is invalid link urls on the about/support page to before-contributing page (might not be in the scope of this PR)

File: new-site/src/docs/components/index.mdx (hunk 2 above)
Comment (ronijaakkola) [reply new-site/src/docs/components/index.mdx:8]: I have been dreaming of a list of component cards. They would include a small image of the component, the component name and maybe the status. Then you could filter that list of cards with an input.

I guess for now we can just add the table too.

File: new-site/src/docs/components/tooltip/code.mdx (hunk 1 above)
Comment (ronijaakkola) [reply new-site/src/docs/components/tooltip/code.mdx]: - [ x] Fixed tabs not rendering on the Tooltip documentation page

File: new-site/src/docs/components/index.mdx (hunk 2 above)
Comment (ronijaakkola) [reply new-site/src/docs/components/index.mdx]: Something like this; maybe with an image and some relevant tags.

image

Comment (minevala) [reply new-site/src/docs/components/index.mdx]: Ok 👍 Should we make a separate task about that?

Comment (ronijaakkola) [reply new-site/src/docs/components/index.mdx]: >  Should we make a separate task about that?
I will add a ticket and add basic specs.

File: new-site/src/docs/foundation/guidelines/checkbox-radiobutton-toggle.mdx [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
+nav_title: 'Checkboxes, radio buttons, or toggles?'
 ---
 
 import { Notification, Checkbox, RadioButton, ToggleButton, SelectionGroup, Link } from "hds-react";
-import LeadParagraph from '../../components/LeadParagraph';
-import PlaygroundPreview from '../../components/Playground';
+import LeadParagraph from '../../../components/LeadParagraph';
+import PlaygroundPreview from '../../../components/Playground';
 
 # Checkboxes, radio buttons, or toggles?
```
Comment (minevala) [inline new-site/src/docs/foundation/guidelines/checkbox-radiobutton-toggle.mdx:12]: Component links of this page are still pointing into elements folder

File: new-site/src/docs/foundation/guidelines/localisation.mdx [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
 title: "Localisation"
+nav_title: "Localisation"
 ---
 
-import LeadParagraph from '../../components/LeadParagraph';
-import Image from '../../components/LeadParagraph';
+import LeadParagraph from '../../../components/LeadParagraph';
+import Image from '../../../components/LeadParagraph';
 
 # Localisation
```
Comment (minevala) [inline new-site/src/docs/foundation/guidelines/localisation.mdx:11]: Navigation component link on this page is still pointing into elements folder

**Review by minevala — COMMENTED** [review]
Nice work 👍 
I noticed that some internal links are still pointing into non-existing elements-folder

File: new-site/src/docs/foundation/guidelines/checkbox-radiobutton-toggle.mdx (hunk 3 above)
Comment (ronijaakkola) [reply new-site/src/docs/foundation/guidelines/checkbox-radiobutton-toggle.mdx:12]: - [ x ] Fixed broken links in the Checkbox/RadioButton/Toggle guideline page

File: new-site/src/docs/foundation/guidelines/localisation.mdx (hunk 4 above)
Comment (ronijaakkola) [reply new-site/src/docs/foundation/guidelines/localisation.mdx:11]: - [ ] Fixed a broken link in the Localisation guideline page

**Review by minevala — APPROVED** [review]
Documentation content is almost there 🎉

**Pull request merged**